migrations/entitlements/migration.go

```diff
@@ -281,10 +281,11 @@ func ConvertValueToEntitlements(
 			return true
 		})
 
-		return interpreter.NewDictionaryValue(
+		return interpreter.NewDictionaryValueWithAddress(
 			inter,
 			interpreter.EmptyLocationRange,
 			dictionaryStaticType,
+			v.GetOwner(),
```

Comment:
Interesting: should we do the same for all places where dictionary values are created? e.g: in migration framework

Reply:
Yeah, I wondered about that too and searched for `NewDictionaryValue*` in `/migrations`. 
However, only the entitlements migration creates new containers, and the cases for other containers (like for arrays) already correctly construct new values with the proper address.